### PR TITLE
Update go module path to k8s.io/kube-state-metrics/v2

### PR DESCRIPTION
**Pull request by omegas27**
Signed-off-by: Quentin Bisson <quentin@giantswarm.io>

**What this PR does / why we need it**:

Updating go module path to v2 towards https://github.com/kubernetes/kube-state-metrics/issues/569

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,20 +1,28 @@
-module k8s.io/kube-state-metrics
+module k8s.io/kube-state-metrics/v2
 
 require (
+	github.com/Azure/go-autorest/autorest v0.10.0 // indirect
```
Comment (lilic) [inline go.mod:4]: Curious of all the changes that were added?

**Review by lilic — COMMENTED** [review]
Thanks for the PR, have one question!

cc @brancz @tariq1890 PTAL

File: go.mod (hunk 1 above)
Comment (omegas27) [reply go.mod:4]: I just triggered `go mod tidy` but maybe I should have done it in a different way?

Comment (lilic) [reply go.mod:4]: It's not a problem at all, mainly asking as we will be porting this to release 2.0 branch and did not want to introduce any new not needed changes, but don't think its a problem here.

**Pull request merged**